b2/src/main/java/org/jclouds/b2/config/B2HttpApiModule.java

```diff
@@ -0,0 +1,77 @@
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.labs.b2.config;
+
+import java.util.concurrent.TimeUnit;
+
+import javax.inject.Singleton;
+
+import org.jclouds.labs.b2.B2Api;
+import org.jclouds.labs.b2.domain.Authorization;
+import org.jclouds.labs.b2.filters.RequestAuthorization;
+import org.jclouds.labs.b2.handlers.ParseB2ErrorFromJsonContent;
+import org.jclouds.http.HttpErrorHandler;
+import org.jclouds.http.annotation.ClientError;
+import org.jclouds.http.annotation.Redirection;
+import org.jclouds.http.annotation.ServerError;
+import org.jclouds.rest.ConfiguresHttpApi;
+import org.jclouds.rest.config.HttpApiModule;
+
+import com.google.common.base.Function;
+import com.google.common.base.Supplier;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.inject.Provides;
+import com.google.inject.Scopes;
+
+/** Configures the mappings. Installs the Object and Parser modules. */
+@ConfiguresHttpApi
+public final class B2HttpApiModule extends HttpApiModule<B2Api> {
+   @Override
+   protected void configure() {
+      super.configure();
+      bind(RequestAuthorization.class).in(Scopes.SINGLETON);
+   }
+
+   @Override
+   protected void bindErrorHandlers() {
+      bind(HttpErrorHandler.class).annotatedWith(Redirection.class).to(ParseB2ErrorFromJsonContent.class);
+      bind(HttpErrorHandler.class).annotatedWith(ClientError.class).to(ParseB2ErrorFromJsonContent.class);
+      bind(HttpErrorHandler.class).annotatedWith(ServerError.class).to(ParseB2ErrorFromJsonContent.class);
+   }
+
+   @Provides
+   @Singleton
+   static Supplier<Authorization> provideAuthorizationCache(final B2Api b2Api) {
+      final Object fakeKey = new Object();  // fake key for single-element cache
+      final LoadingCache<Object, Authorization> cache = CacheBuilder.newBuilder()
+            .expireAfterWrite(23, TimeUnit.HOURS)
+            .build(CacheLoader.from(new Function<Object, Authorization>() {
+               @Override
+               public Authorization apply(Object nothing) {
+                  return b2Api.getAuthorizationApi().authorizeAccount();
+               }
+            }));
+      return new Supplier<Authorization>() {
+            @Override
+            public Authorization get() {
+               return cache.getUnchecked(fakeKey);
+            }
+         };
+   }
```

Comment:
A common practice in jclouds when you are providing a supplier on top of a cache and binding it to the Guice contex, is to use the jclouds memoized supplier. It already takes care of properly dealing with auth exceptions. It would be something like:
```java
@Provides
@Singleton
protected final Supplier<Authorization> provideAuthorizationSupplier(final B2Api b2Api) {
   return new Supplier<Authorization>() {
      @Override public Authorization get() {
         return b2Api.getAuthorizationApi().authorizeAccount();
      }
   };
}

@Provides
@Singleton
@Memoized
protected final Supplier<Authorization> provideAuthorizationCache(
      AtomicReference<AuthorizationException> authException,
      @Named(PROPERTY_SESSION_INTERVAL) long seconds,
      Supplier<Authorization>  uncached) {
   return MemoizedRetryOnTimeOutButNotOnAuthorizationExceptionSupplier.create(
      authException, uncached, seconds, TimeUnit.SECONDS);
```

With this you can inject both suppliers. If you want to inject the cache, you just need to qualify the variable as `@Memoized`.

Also note that the `@Provides` methods are better marked final, as Guice 4 does not allow to override them. I've also used the `PROPERTY_SESSION_INTERVAL` property to configure the auth expiration. Feel free to consider if that makes sense in this provider. If it does, you could configure the 23 hours default value for that property in the api metadata.